sherpa-onnx/csrc/online-transducer-greedy-search-nemo-decoder.cc

```diff
@@ -0,0 +1,198 @@
+
+  decoder_states = std::move(decoder_output_pair.second);
+
+  // TODO: Inside this loop, I need to framewise decoding.
+  for (int32_t t = 0; t != num_rows; ++t) {
+    Ort::Value cur_encoder_out = Ort::Value::CreateTensor(
+        memory_info, const_cast<float *>(encoder_out) + t * num_cols, num_cols,
+        encoder_shape.data(), encoder_shape.size());
+
+    Ort::Value logit = model->RunJoiner(std::move(cur_encoder_out),
+                                         View(&decoder_output_pair.first));
+
+    float *p_logit = logit.GetTensorMutableData<float>();
+    if (blank_penalty > 0) {
+      p_logit[blank_id] -= blank_penalty;
+    }
+
+    auto y = static_cast<int32_t>(std::distance(
+        static_cast<const float *>(p_logit),
+        std::max_element(static_cast<const float *>(p_logit),
+                         static_cast<const float *>(p_logit) + vocab_size)));
+    SHERPA_ONNX_LOGE("y=%d", y);
+    if (y != blank_id) {
+      r.tokens.push_back(y);
+      r.timestamps.push_back(t + r.frame_offset);
+
+      decoder_input_pair = BuildDecoderInput(y, model->Allocator());
+
+      // last decoder state becomes the current state for the first chunk
+      decoder_output_pair =
+          model->RunDecoder(std::move(decoder_input_pair.first),
+                             std::move(decoder_states));
+
+      // Update the decoder states for the next chunk
+      decoder_states = std::move(decoder_output_pair.second);
+    }
+  }
+
+  decoder_out = std::move(decoder_output_pair.first);
+//  UpdateCachedDecoderOut(model->Allocator(), &decoder_out, result);
+
+  // Update frame_offset
+  for (auto &r : *result) {
+    r.frame_offset += num_rows;
+  }
+
+  return std::move(decoder_states);
+}
+
+
+std::vector<Ort::Value> OnlineTransducerGreedySearchNeMoDecoder::Decode(
+    Ort::Value encoder_out, 
+    std::vector<Ort::Value> decoder_states,
+    std::vector<OnlineTransducerDecoderResult> *result,
+    OnlineStream ** /*ss = nullptr*/, int32_t /*n= 0*/) {
+
+  auto shape = encoder_out.GetTensorTypeAndShapeInfo().GetShape();
+
+  if (shape[0] != result->size()) {
+    SHERPA_ONNX_LOGE(
+        "Size mismatch! encoder_out.size(0) %d, result.size(0): %d",
+        static_cast<int32_t>(shape[0]),
+        static_cast<int32_t>(result->size()));
+    exit(-1);
+  }
+
+  int32_t batch_size = static_cast<int32_t>(shape[0]);  // bs = 1
+  int32_t dim1 = static_cast<int32_t>(shape[1]); // 2
+  int32_t dim2 = static_cast<int32_t>(shape[2]); // 512
+
+  // Define and initialize encoder_out_length
+  Ort::MemoryInfo memory_info = Ort::MemoryInfo::CreateCpu(OrtDeviceAllocator, OrtMemTypeCPU);
+  
+  int64_t length_value = 1;
+  std::vector<int64_t> length_shape = {1};
+  
+  Ort::Value encoder_out_length = Ort::Value::CreateTensor<int64_t>(
+      memory_info, &length_value, 1, length_shape.data(), length_shape.size()
+  );
+
+  const int64_t *p_length = encoder_out_length.GetTensorData<int64_t>();
```

Comment:
```suggestion
```

I hope you can understand that for the streaming case, the encoder_out_length is `dim1`.